src/generators/Generator.js

```diff
@@ -260,14 +260,26 @@ export default class Generator {
 					// export is last property, we can just return it
 					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `return ` );
 				} else {
-					// TODO ensure `template` isn't already declared
-					this.code.overwrite( defaultExport.start, defaultExport.declaration.start, `var template = ` );
+					// we need to avoid a conflict with anything on the top-level scope of the component
+					// however, determining which things these are is tricky, so we instead just avoid conflicts with any identifiers anywhere
```

Comment:
I think it might be possible to use [`annotateWithScopes`](https://github.com/sveltejs/svelte/blob/master/src/generators/annotateWithScopes.js) to only get top-level identifiers — the returned top-level `scope` object will have a `declarations` property containing anything declared in the top level (or `var` declarations in non-function blocks)